homeassistant/components/climate/netatmo.py

```diff
@@ -0,0 +1,148 @@
+    """Setup the NetAtmo Thermostat."""
+    netatmo = get_component('netatmo')
+    device = config.get(CONF_RELAY)
+    away_temp = config.get(CONF_AWAY_TEMPERATURE)
+    import lnetatmo
+    try:
+        data = ThermostatData(netatmo.NETATMO_AUTH, device)
+    except lnetatmo.NoDevice:
+        return None
+
+    for module_name in data.get_module_names():
+        if CONF_THERMOSTAT in config:
+            if config[CONF_THERMOSTAT] != [] and \
+               module_name not in config[CONF_THERMOSTAT]:
+                continue
+
+        add_callback_devices([NetatmoThermostat(data, module_name)])
+
+
+class NetatmoThermostat(ClimateDevice):
+    """Representation a Netatmo thermostat."""
+
+    def __init__(self, data, module_name, away_temp=None):
+        """Initialize the sensor."""
+        self._data = data
+        self._state = None
+        self._name = module_name
+        self._target_temperature = None
+        self._away_temp = away_temp
+        self._away = None
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._data.thermostatdata.temp
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._data.current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def current_operation(self):
+        """Return the current state of the thermostat."""
+        state = self._data.thermostatdata.relay_cmd
+        if state == 0:
+            return STATE_IDLE
+        elif state == 100:
+            return STATE_HEAT
+
+    @property
+    def is_away_mode_on(self):
+        """Return true if away mode is on."""
+        return self._away
+
+
+    def turn_away_mode_on(self):
+        """Turn away on."""
+        mode = "away"
+        temp = None
+        self._data.thermostatdata.setthermpoint(mode, temp, endTimeOffset=None)
+        self._away = True
+        self.update_ha_state()
+
+    def turn_away_mode_off(self):
+        """Turn away off."""
+        mode = "program"
+        temp = None
+        self._data.thermostatdata.setthermpoint(mode, temp, endTimeOffset=None)
+        self._away = False
+        self.update_ha_state()
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+        mode = "manual"
+        endTimeOffset = 7200
+        self._data.thermostatdata.setthermpoint(mode, temperature, endTimeOffset)
```

Comment:
In Python, you don't need to put a backslash inside bracket if you want to split a line (it will know that the line continue until the closing bracket), but you need to put a backslash elsewhere to say that you're splitting the line